substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/JavaMainWrapper.java

```diff
@@ -287,3 +295,16 @@ public static void enter(int paramArgc, CCharPointerPointer paramArgv) {
         void setArgv(CCharPointerPointer value);
     }
 }
+
+@TargetClass(className = "java.lang.Terminator")
+final class Target_java_lang_Terminator {
+    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
+    private static Target_jdk_internal_misc_Signal_Handler handler;
```

Comment:
For this to work on JDK 11 (`handler` field changed type) you could declare it as:
```suggestion
    private static Object handler;
```

Reply:
Nahh, that doesn't work.
```
Error: Type mismatch:
    class java.lang.Object private static java.lang.Object com.oracle.svm.core.Target_java_lang_Terminator.handler
    interface jdk.internal.misc.Signal$Handler private static jdk.internal.misc.Signal$Handler java.lang.Terminator.handler
```
Instead it's much easier to use: https://github.com/oracle/graal/pull/2393/commits/6bebeb45724956f33357ff14caf63fec5757f0ce